test/errors.test.ts

```diff
@@ -13,55 +13,56 @@ const tsErrors = (code: string) => {
 
 test('toEqualTypeOf<...>() error message', async () => {
   expect(tsErrors(`expectTypeOf({a: 1}).toEqualTypeOf<{a: string}>()`)).toMatchInlineSnapshot(`
-    "test/test.ts:3:22 - error TS2554: Expected 1 arguments, but got 0.
+    "test/test.ts:3:36 - error TS2344: Type '{ a: string; }' does not satisfy the constraint 'Mismatch<{ a: number; }>'.
+      Type '{ a: string; }' is not assignable to type '{ a: number; }'.
+        Types of property 'a' are incompatible.
+          Type 'string' is not assignable to type 'number'.
 
     3 expectTypeOf({a: 1}).toEqualTypeOf<{a: string}>()
-                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-      src/index.ts:118:16
-        118     <Expected>(...MISMATCH: MismatchArgs<Equal<Actual, Expected>, B>): true
-                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-        Arguments for the rest parameter 'MISMATCH' were not provided.
+                                         ~~~~~~~~~~~
     "
   `)
 })
 
 test('toEqualTypeOf(...) error message', async () => {
   expect(tsErrors(`expectTypeOf({a: 1}).toEqualTypeOf({a: 'one'})`)).toMatchInlineSnapshot(`
-    "test/test.ts:3:36 - error TS2345: Argument of type '{ a: string; }' is not assignable to parameter of type 'never'.
+    "test/test.ts:3:37 - error TS2322: Type 'string' is not assignable to type 'number'.
 
     3 expectTypeOf({a: 1}).toEqualTypeOf({a: 'one'})
-                                         ~~~~~~~~~~
+                                          ~
+
+      test/test.ts:3:15
+        3 expectTypeOf({a: 1}).toEqualTypeOf({a: 'one'})
+                        ~~~~
+        The expected type comes from property 'a' which is declared here on type 'Mismatch<{ a: number; }>'
     "
   `)
 })
 
 test('toMatchTypeOf<...>() error message', async () => {
   expect(tsErrors(`expectTypeOf({a: 1}).toMatchTypeOf<{a: string}>()`)).toMatchInlineSnapshot(`
-    "test/test.ts:3:22 - error TS2554: Expected 1 arguments, but got 0.
+    "test/test.ts:3:36 - error TS2344: Type '{ a: string; }' does not satisfy the constraint 'Mismatch<{ a: number; }>'.
+      Type '{ a: string; }' is not assignable to type '{ a: number; }'.
+        Types of property 'a' are incompatible.
+          Type 'string' is not assignable to type 'number'.
```

Comment:
This error message appears backwards to me.

`expectTypeOf<Actual>().toMatchTypeOf<Expected>`

So I'd expect it to be:

> `Type 'number' is not assignable to type 'string'`

etc.